kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -80,21 +86,27 @@ def __init__(self, name, init_config, agentConfig, instances=None):
             'kube_service_spec_type': self.count_objects_by_tags,
         }
 
+        # Handling jobs succeeded/failed counts
+        self.failed_job_counts = defaultdict(KubernetesState.JobCount)
+        self.succeeded_job_counts = defaultdict(KubernetesState.JobCount)
+
     def check(self, instance):
         endpoint = instance.get('kube_state_url')
 
-        # Job counters are monotonic: they increase at every run of the job
-        # We want to send the delta via the `monotonic_count` method
-        self.job_succeeded_count = defaultdict(int)
-        self.job_failed_count = defaultdict(int)
-
         scraper_config = self.config_map[endpoint]
         self.process(scraper_config, metric_transformers=self.METRIC_TRANSFORMERS)
 
-        for job_tags, job_count in iteritems(self.job_succeeded_count):
-            self.monotonic_count(scraper_config['namespace'] + '.job.succeeded', job_count, list(job_tags))
-        for job_tags, job_count in iteritems(self.job_failed_count):
-            self.monotonic_count(scraper_config['namespace'] + '.job.failed', job_count, list(job_tags))
+        for job_tags, job in iteritems(self.failed_job_counts):
+            self.monotonic_count(scraper_config['namespace'] + '.job.failed', job.count, list(job_tags))
+            if job.current_run_max_ts > 0:
+                job.previous_run_max_ts = job.current_run_max_ts
+                job.current_run_max_ts = 0
+
+        for job_tags, job in iteritems(self.succeeded_job_counts):
+            self.monotonic_count(scraper_config['namespace'] + '.job.succeeded', job.count, list(job_tags))
+            if job.current_run_max_ts > 0:
+                job.previous_run_max_ts = job.current_run_max_ts
+                job.current_run_max_ts = 0
```

Comment:
Could be simplified to a method on `JobCount`:
```
            class JobCount:
                def reset_current_run_max(self):
                    if self.current_run_max_ts > 0:
                        self.previous_run_max_ts = self.current_run_max_ts
                        self.current_run_max_ts = 0
```

Then used in above two locations.